[40기 김채영]Fix: conflict 해결 완료

kimcodud opened the pull request:
##::TRELLO TICKET
https://trello.com/c/4U2kVgQx
해당 PR은 https://trello.com/c/y7HddaIt와 따라감


- nav bar 왼쪽 하단 카테고리 탭 hover 시 메인메뉴 카테고리 컴포넌트 구현
- 메인메뉴와 서브 카테고리 id 일치 시 서브 카테고리 컴포넌트 구현

## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [x] 리뷰 반영
- [ ] 리팩토링
- [X] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- nav bar 하단 카테고리 컴포넌트 분리하여 드롭다운 기능이 있는 카테고리 컴포넌트 구현
- 카테고리 메인메뉴의 서브카테고리 유무에 따라 추가 상세 메뉴 구현

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
- 카테고리 관련 mock data 생성하여  CategoryList 컴포넌트 안에서 map함수 이용해 카테고리 리스트 구현
- return 문 내부에서 다시한번 subcategory 와 state 로 저장한 초기 id 일치 시 subcategory 추가 구현

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- mock data를 활용하여 데이터를 받아와 map으로 그리는 방식에 대해 이해하게 됨

<br />

## :: 기타 질문 및 특이 사항

- 제가 구현하고자 하는 카테고리 스타일은 아래 사진과 같은데
<img width="555" alt="스크린샷 2022-12-12 오후 2 05 34" src="https://user-images.githubusercontent.com/109934186/206965126-ffc66048-b6bc-4b75-ad81-93d2a4b498b5.png">
- 메인 카테고리 리스트에 hover 효과를 주면 아래 사진과 같이 서브카테고리까지 같은 효과가 들어갑니다. 혹시 서브 카테고리에서는 메인 카테고리에서 적용한 스타일을 제외할 수 있을까요..?
<img width="989" alt="스크린샷 2022-12-12 오후 2 11 12" src="https://user-images.githubusercontent.com/109934186/206965476-797375ff-eb54-4d0c-9def-e3ff179cd7ee.png">

<img width="638" alt="스크린샷 2022-12-12 오후 2 07 36" src="https://user-images.githubusercontent.com/109934186/206965145-bf43da1c-e9f5-49bd-9a2e-674d7ab185fc.png">



> > 메인 카테고리 리스트와 서브 카테고리 리스트에 mouse를 hover 했을 때 각기 다른 효과를 주고 싶은데 메인 카테고리 리스트에 적용한 style이 서브 카테고리에도 함께 적용이 됩니다. 혹시 따로따로 style을 적용할 수 있는 방법이 있을지 궁금합니다.
> 
> 혹시 이부분을 알 수 있는 캡쳐화면이 있다면 아래 코멘트에 남겨주세요! 아래 화면에서 어디영역인지 알고 싶습니다!
> 
> ![image](https://user-images.githubusercontent.com/53929065/206950235-1cea049f-2325-4aa7-8ef9-bc3a098ba094.png)

kimcodud committed:
Add: nav bar 드롭다운 카테고리 기능 구현
- nav bar 왼쪽 하단 카테고리 탭 hover 시 메인메뉴 카테고리 컴포넌트 구현
- 메인메뉴와 서브 카테고리 id 일치 시 서브 카테고리 컴포넌트 구현

YunKukPark commented:
> 메인 카테고리 리스트와 서브 카테고리 리스트에 mouse를 hover 했을 때 각기 다른 효과를 주고 싶은데 메인 카테고리 리스트에 적용한 style이 서브 카테고리에도 함께 적용이 됩니다. 혹시 따로따로 style을 적용할 수 있는 방법이 있을지 궁금합니다.

혹시 이부분을 알 수 있는 캡쳐화면이 있다면 아래 코멘트에 남겨주세요! 
아래 화면에서 어디영역인지 알고 싶습니다! 

![image](https://user-images.githubusercontent.com/53929065/206950235-1cea049f-2325-4aa7-8ef9-bc3a098ba094.png)

YunKukPark commented on public/data/categoryList.json:
```diff
@@ -0,0 +1,31 @@
+    "id": 2,
+    "icon": "/images/pill.png",
+    "mainname": "영양제",
+    "subcategory": [
```
mock data 활용하기 참 좋은 형태로 잘 만들어주셨군요.. ✨✨✨
눈부셔서 코드를 볼 수가 없네요

kimcodud replied:
🥹 감사합니답!!

kimcodud replied:
@YunKukPark 윤국님! 리뷰해주신 부분들 수정하였는데 혹시 바로 머지가 가능하다면 추가질문사항은 나중에 구현해도 될까요?

YunKukPark replied:
@kimcodud 넵 물론이죠! 
그치만 얄궂은 컨플릭트가 나버렸는걸요.. 🥲
컨플릭트만 해결해주세요!

YunKukPark commented on src/Router.js:
```diff
@@ -10,6 +13,9 @@ import ProductList from './pages/ProductList/ProductList';
     <BrowserRouter>
+      <Nav />
+      <MainCarousel />
+      <ProductCarousel />
```
이렇게 하면 실 적용 되었을 때에,
모든 페이지에서 MaiCarousel과 ProductCarousel을 볼 수 있습니다. 
Test 가 끝나면 지워주세요!

YunKukPark commented on src/components/CategoryList/CategoryList.js:
```diff
@@ -0,0 +1,53 @@
+                  <img key={id} src={icon} alt={mainname} />
+                </div>
+                <div className="mainNameWrapper">{mainname}</div>
+                {currentId === id && subcategory && (
+                  <ul className="categoryListSub">
+                    {subcategory.map(({ id, name }) => {
+                      return (
+                        <li className="subCategory" key={id} data={name}>
+                          {name}
+                        </li>
+                      );
+                    })}
+                  </ul>
+                )}
```
이 boolean 값은 따로 빼주는게 가독성에 더 좋아보여요! 
`const isCurrentTab = ....`

YunKukPark commented on src/components/CategoryList/CategoryList.scss:
```diff
@@ -0,0 +1,68 @@
+@import '../../styles/mixins/flex.scss';
+$primary-color: #00388d;
```
styles/variables.scss 에서 공용으로 color에 대한 관리가 이뤄졌으면 좋겠는데 어떻게 생각하시나요?

YunKukPark commented on src/components/CategoryList/CategoryList.js:
```diff
@@ -0,0 +1,53 @@
+                className="mainCategory"
+                key={id}
+                data={mainname}
+                onMouseEnter={() => setCurrentId(id)}
```
이런 callback은 커링을 이용하면 정말 이쁘게 표현 할 수 있습니다.

```jsx
const onMouseEnterTab = (id) => (e) => setCurrentId(id);
```
```suggestion
                onMouseEnter={setCurrentId(id)}
```

callback 함수는 들어오는 argument와 parameter가 같다면 생략이 가능하다는 점을 이용한 겁니다.

이걸 참고하면 이해가 될 것 같습니다. 
[참고자료](https://www.youtube.com/watch?v=dhrXFrrkYf0)

YunKukPark reviewed (commented):
너무 좋습니다 👍🏻👍🏻👍🏻

도움요청으로 올려주신 글은 아직 잘 파악이 되지 않아서 스크린샷과 함께 남겨주시면 제가 이해하는데 도움이 더 될 것 같아요!! 

추가적인 코멘트를 남겨두었으니 확인해주시면 감사하겠습니다 🙇🏻‍♂️🙇🏻‍♂️

kimcodud committed:
Modify: 리뷰 반영 수정
- Router.js 경로 삭제
- 옵셔널 체이닝 이용해 boolean 값에 접근하도록 수정
- styles/variables.scss 에 공용으로 관리하도록  color 변수 입력
- argument 와 parameter 가 같은 callback 함수 수정

kimcodud committed:
Fix: conflict 해결

kimcodud commented:
@YunKukPark conflict 해결완료했습니다!

kimcodud committed:
Modify: Router.js 경로 수정

YunKukPark reviewed (approved):
고생하셨습니다 채영님. 
다음티켓 부시러가시죠 🚀🚀🚀
